src/machine/machine_rp2040_adc.go

```diff
@@ -20,43 +39,112 @@ func InitADC() {
 	waitForReady()
 }
 
-// Configure configures a ADC pin to be able to be used to read data.
+// Below methods are for standard tinygo, ADC{} pin based access to the ADC peripheral.
+
+// The Configure method sets the ADC pin to analog input mode.
+// Note the ADCConfig parameter is ignored.
+// Does nothing if the GPIO is not an ADC muxed pin.
 func (a ADC) Configure(config ADCConfig) {
+	if c, ok := a.GetADCChannel(); ok {
+		c.Configure(ADCChConfig{})
+	}
+}
+
+// The Get method returns a one-shot ADC sample reading.
+func (a ADC) Get() uint16 {
+	if c, ok := a.GetADCChannel(); ok {
+		return c.GetOnce()
+	}
+	// Not an ADC pin!
+	return 0
+}
+
+// The GetADCChannel method returns the channel associated with the ADC pin.
+// Returns !ok if the pin is not an ADC pin
+func (a ADC) GetADCChannel() (c ADCChannel, ok bool) {
+	ok = true
 	switch a.Pin {
-	case ADC0, ADC1, ADC2, ADC3:
-		a.Pin.Configure(PinConfig{Mode: PinAnalog})
+	case ADC0:
+		c = ADC0_CH
+	case ADC1:
+		c = ADC1_CH
+	case ADC2:
+		c = ADC2_CH
+	case ADC3:
+		c = ADC3_CH
 	default:
-		// invalid ADC
-		return
+		// Invalid ADC pin
+		ok = false
 	}
+	return c, ok
 }
 
-func (a ADC) Get() uint16 {
-	rp.ADC.CS.SetBits(uint32(a.getADCChannel()) << rp.ADC_CS_AINSEL_Pos)
+// Below methods are rp2040 specific, channel based access to the ADC peripheral.
+// The adcChannel methods provide access to the temperature sensor. As well as the pin based channels.
+
+// The Configure method sets the channel's associated pin to analog input mode.
+// Configuring the ADC_TEMP_CH channel will power on the temperature sensor.
+// The powered on temperature sensor increases ADC_AVDD current by approximately 40 μA.
+func (c ADCChannel) Configure(config ADCChConfig) {
+	if p, ok := c.Pin(); ok {
+		p.Configure(PinConfig{Mode: PinAnalog})
+	}
+	if c == ADC_TEMP_CH {
+		// Enable temperature sensor bias source
+		rp.ADC.CS.SetBits(rp.ADC_CS_TS_EN)
+	}
+}
+
+// The GetOnce method returns a one-shot ADC sample reading
+func (c ADCChannel) GetOnce() uint16 {
```

Comment:
Not sure about exposing this as public function, due to not existing on other boards. Perhaps for now it should be internal package function only?

Reply:
Will change to not private. This was a way to get the raw temperature sensor ADC reading if using an external ref voltage less than 3.3V. I'll update code to use ADCConfig.Reference (if set) when calculating the voltage and temperature.